src/Features/Core/Portable/AddParameter/AbstractAddParameterCodeFixProvider.cs

```diff
@@ -113,125 +137,269 @@ internal abstract class AbstractAddParameterCodeFixProvider<
-                var signature = $"{type.Name}({string.Join(", ", parameters)})";
+                if (hasCascadingDeclarations)
+                {
+                    var titleForCascadingFix = GetCodeFixTitle(FeaturesResources.Add_parameter_to_0_including_overrides_implementations, methodToUpdate, parameters);
+                    context.RegisterCodeFix(new CodeAction.CodeActionWithNestedActions(title, ImmutableArray.Create<CodeAction>(
+                            new MyCodeAction(title,
+                                c => FixAsync(context.Document, methodToUpdate, argumentToInsert, arguments, fixAllReferences: false, c)),
+                            new MyCodeAction(titleForCascadingFix,
+                                c => FixAsync(context.Document, methodToUpdate, argumentToInsert, arguments, fixAllReferences: true, c))),
+                            isInlinable: true),
+                        context.Diagnostics);
+                }
+                else
+                {
+                    context.RegisterCodeFix(
+                        new MyCodeAction(title,
+                            c => FixAsync(context.Document, methodToUpdate, argumentToInsert, arguments, fixAllReferences: false, c)),
+                        context.Diagnostics);
+                }
+            }
+        }
 
-                var title = string.Format(FeaturesResources.Add_parameter_to_0, signature);
+        /// <summary>
+        /// Checks if there are indications that there might be more than one declarations that need to be fixed.
+        /// The check does not look-up if there are other declarations (this is done later in the CodeAction).
+        /// </summary>
+        private bool HasCascadingDeclarations(IMethodSymbol method)
+        {
+            // Don't cascade constructors
+            if (method.IsConstructor())
+            {
+                return false;
+            }
+
+            // Virtual methods of all kinds might have overrides somewhere else that need to be fixed.
+            if (method.IsVirtual || method.IsOverride || method.IsAbstract)
+            {
+                return true;
+            }
 
-                context.RegisterCodeFix(
-                    new MyCodeAction(title, c => FixAsync(document, constructor, argumentToAdd, arguments, c)),
-                    context.Diagnostics);
+            // If interfaces are involved we will fix those too
+            // Explicit interface implementations are easy
+            if (method.ExplicitInterfaceImplementations.Length > 0)
+            {
+                return true;
             }
+
+            // For implicit interface implementations lets check if the characteristic of the method
+            // allows it to implicit implement an interface member.
+            if (method.DeclaredAccessibility == Accessibility.Private || method.DeclaredAccessibility == Accessibility.NotApplicable)
```

Comment:
I'm not certain about the protected variations (Protected, ProtectedAndInternal, ProtectedOrInternal). Please tell me what modifiers to add here.

Reply:
Basically, the method must be public (in C#) for it to be an implicit interface implementation.  VB has no implicit interface impl.  All interface impl in VB is explicit.

Reply:
Internal methods can implement internal interfaces, don't they? What about protected internal combinations then?

Reply:
Nope.  All implicit interface implementation in c# must be public.  Check it out :)